### PR TITLE
test/deck

### PR DESCRIPTION
**Pull request by Segelzwerg**

### CODE REVIEW

File: whist/core/deck.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def add(self, card: Card) -> NoReturn:
         :param card: card to add
         """
         if card in self.__cards:
-            raise KeyError
+            raise KeyError(f'{card.name} already in deck')
```
Comment (iTitus) [inline whist/core/deck.py:32]:
```suggestion
            raise KeyError(f'{card} already in deck')
```
__str__ is implemented

**Review by iTitus — CHANGES_REQUESTED** [review]

File: tests/whist/core/test_deck.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        self.deck.add(self.spades_king)
+        with self.assertRaises(KeyError):
+            self.deck.add(self.spades_king)
+
+    def test_iter(self):
+        queen_diamonds = Card(Suit.DIAMONDS, Rank.Q)
+        cards = [self.spades_king, queen_diamonds]
+        deck = Deck(cards)
+        for expected, card in zip(cards, deck):
+            self.assertEqual(expected, card)
```
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply tests/whist/core/test_deck.py:41]: indeed, any suggestion on how we could test for iter without considering the order?

File: whist/core/deck.py (hunk 1 above)
Comment (Segelzwerg) [reply whist/core/deck.py:32]: I wasn't sure if it works in `f'{}'`

File: tests/whist/core/test_deck.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (iTitus) [reply tests/whist/core/test_deck.py:41]: - dont test `iter`
- convert to set and use `assertEqual`
- `assertEqualInAnyOrder` ?

**Review by iTitus — APPROVED** [review]

**Pull request merged**